### ISSUE TITLE
navbar inconsistent behaviour 

### ISSUE THREAD

**Issue by tassoman** [vuejs/vitepress#2085]
### Describe the bug

Navbar representation is inconsistent between components, hovering them, they produce three different reactions

![vp1](https://user-images.githubusercontent.com/92727/224983475-11ee69b7-03a8-437f-84b6-2f4e71f7454d.png)

Hovering a nav menu link, produces fade to brand color (with no underline for accessibility)

![vp2](https://user-images.githubusercontent.com/92727/224983483-3d5b4682-b0c3-4c84-a11e-f90dbf6017bb.png)

Hovering socialLinks components produces more contrast, but it's an outgoing link. 

![vp3](https://user-images.githubusercontent.com/92727/224983489-df4c0996-b609-4109-ae66-4e70169099cf.png)

Hovering the title, produce a less contrast. Again, here it's a link should have brand color and underline for accessibility. Lower contrast is not WCAG Level 2 compliant. It happens also for dropdown menu anchors.

### Reproduction

Build basic website having this attributes

```typescript
        socialLinks: [
            { icon: 'github', link: 'https://github.com/'},
        ],
        nav: [
            {
                text: 'dropdown menu',
                items: [
                    {
                        text: 'one', 
                        link: 'https://bing.it/',
                        target: '_blank',
                    },{ 
                        text: 'two',
                        link: 'https://google.it/',
                    },
                ]
            },
            { 
                text: 'internal link', 
                link: 'mypage',
            },
      ],
```

### Expected behavior

I expect underlined every link turning into brand color no matter component is.

### System Info

```shell
System:
    OS: Linux 5.10 Ubuntu 22.04.2 LTS 22.04.2 LTS (Jammy Jellyfish)
    CPU: (8) x64 Intel(R) Core(TM) i5-10210U CPU @ 1.60GHz
    Memory: 11.40 GB / 12.31 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 16.15.1 - ~/.nvm/versions/node/v16.15.1/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.15.1/bin/yarn
    npm: 8.19.2 - ~/.nvm/versions/node/v16.15.1/bin/npm
  npmPackages:
    vitepress: ^1.0.0-alpha.56 => 1.0.0-alpha.56
```

### Validations

- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by MarkusKeck** [comment]
@tassoman The only behavoir which I would rate as inconsistent is the hovering of a dropdown menu item, where the item gets less contrast while other menu options are highlighted with the brand color
![image](https://user-images.githubusercontent.com/19492739/225073295-a3950678-d6b0-40d3-9392-ccf33e6cdfd5.png)

Most websites don't use an underscore to highlight your menu items when hovering, nor do I think increasing the contrast of the socialLinks is a problem. What do you think should happen here, should the icons also be highlighted in the brand color?

About the brand icon issue, many websites also use a slight increase/decrease in contrast or even no change at all but just changing the cursor to a pointer.

**Comment by yyx990803** [comment]
- The title is different from normal links. It serves as a logotype and doesn't necessarily need to behave like normal links. Even in WCAG, logotype is explicitly exempt from contrast requirements.

- The dropdown itself is not a link. The actual links under the dropdown are consistent with brand color. I also don't see where WCAG mandates links cannot decrease contrast on hover (as long as the hover state still meets minimum contrast requirement)

**Comment by tassoman** [comment]
Actually, dark and light mode contrast between background and text isn't at 100% ratio.  
So maybe there's room to fade to higher contrast (0% vs 100%) when item is focused/hovered and when editor wants to give it importance with strong / enphasis.

The logo hover animation, I'd prefer can ease out abt 10% instead of fading.

**Comment by yyx990803** [comment]
IMO these are mostly your personal preferences. We will try our best to adhere to WCAG standards where things are clearly defined, but I'm afraid we don't have the responsibility to tweak everything to make you happy.

**Comment by tassoman** [comment]
Except two small accessibility issues I'm going to file, the project, overall is pretty planned for inclusion and it's resources efficient. About Logotype, WCAG2AA says it's an [essential part of content](https://www.w3.org/TR/WCAG21/#images-of-text), but nothing more about animation. As you noted, personal tweaks can be extended into a customized personal theme.

Thank you for the effort, although my third notation maybe is unwanted, the first twos, about socialink and dropdown navi-items, I feel can improve the default theme layout.
